adr/0016-compliance-in-pepr.md

```diff
@@ -0,0 +1,51 @@
+# 14. Compliance in Pepr 
+
+Date: 2025-02-13
+
+
+## Status
+
+Pending
+
+
+## Context
+
+Pepr needs to generate reports for compliance purposes, demonstrating adherence to required security and compliance standards through specific bindings. To achieve this, it’s essential to map compliance controls to Pepr’s bindings. During our discovery process, we determined that Pepr’s role in compliance should focus on reporting rather than enforcement. This is because compliance controls are already enforced through the bindings themselves, and Pepr’s primary function is to report on the compliance status of resources.
+
+## Decision
+
+Extend the fluent API with a new keyword function that accepts a strongly typed control, allowing it to be associated with a given binding. This enables users to map compliance controls to Pepr’s bindings, facilitating the generation of compliance reports based on control status.
+
+## Goals
+
+- Pepr to provide the mechanism to run the validation reports.
+- The report generation should be either on-demand or on schedule.
+
+## Non-Goals
```

Comment:
Are there any intended boundaries as goals or non-goals? IE is this specifically for Kubernetes or would there be intent to support infrastructure and other endpoints?

Reply:
Great point, it is a goal to be able to report on non-Pepr governance/non-Kubernetes governance should the module author desire to